test/FunctionalTests/Tests/DataServices/UnitTests/Client.TDD.Tests/Tests/CustomizedHttpWebRequestMessage.cs

```diff
@@ -44,5 +47,20 @@ public override IODataResponseMessage GetResponse()
                     return new MemoryStream(byteArray);
                 });
         }
+
+        public override IAsyncResult BeginGetResponse(AsyncCallback callback, object state)
+        {
+            // using this as APM was deprecated in.net core
+            return Task.Run((() => callback.Invoke(Task.Run((() =>
+            {
+                // simplified as in the EndResponse we ignore the return of this method
+            })))));

```

Comment:
What's going on here? if you're not going to run anything and you only need an IAsyncResult. Why not just

```c#
return Task.CompletedTask;
```

Reply:
Resolved using Task.FromResult() as Task.CompletedTask is not available in 4.5